arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,720 @@
+          },
+          {
+            "name": "movePayload",
+            "description": "The move request payload.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MoveCommitmentAssociationRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CommitmentAssociation"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/commitmentPlans/{planName}": {
+      "get": {
+        "tags": [
+          "CommitmentPlans"
+        ],
+        "operationId": "CommitmentPlans_Get",
+        "description": "Retrieve an Azure ML commitment plan by its subscription, resource group and name.",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CommitmentPlan"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "CommitmentPlans"
+        ],
+        "operationId": "CommitmentPlans_CreateOrUpdate",
+        "description": "Creates a new Azure ML commitment plan resource or updates an existing one.",
+        "parameters": [
+          {
+            "name": "createOrUpdatePayload",
+            "description": "The payload to create or update the Azure ML commitment plan.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CommitmentPlan"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CommitmentPlan"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/CommitmentPlan"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "CommitmentPlans"
+        ],
+        "operationId": "CommitmentPlans_Remove",
+        "description": "Remove an existing Azure ML commitment plan.",
+        "responses": {
+          "200": {
+            "description": "OK"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "CommitmentPlans"
+        ],
+        "operationId": "CommitmentPlans_Patch",
+        "description": "Patch an existing Azure ML commitment plan resource.",
+        "parameters": [
+          {
+            "name": "patchPayload",
+            "description": "The payload to patch the Azure ML commitment plan with.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CommitmentPlan"
```

Comment:
The nuances between a put and a patch are understandable. There is a question
- Are all the properties of the CommitmentPlan updatable in the PATCH operation?
If not, then how is the customer supposed to know what can or cannot be updated without trial and error?
This should be documented in the description of the body parameter or may be create a different model for body parameter of the PATCH operation that only describes the updatable properties.

Trial and Error approach is very frustrating for the end user.